hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/table/AbstractTable.java

```diff
@@ -258,11 +256,36 @@ protected void onBaseEntityEvent(final BaseUIEntityEvent<E> event) {
         return ids;
     }
 
+    /**
+     * Finds the entity object of the given entity ID by performing a database
+     * search
+     * 
+     * @param lastSelectedId
+     *            ID of the entity
+     * @return entity object as Optional
+     */
     protected abstract Optional<E> findEntityByTableValue(I lastSelectedId);
 
-    protected abstract void publishEntityAfterValueChange(E selectedLastEntity);
+    /**
+     * Is called inside the onValueChange()-method and is performed after
+     * selecting the current entity in the table. If it is necessary to publish
+     * one or more events after selecting the entity, this method can be
+     * overridden
+     */
+    protected void publishAdditionalEvent() {
```

Comment:
There are serveral issues here:
1.   Don't express the contract within a comment. 
      This can change very fast and nobody will adapt this comment!   
2.  Don't tell the implementor what he should do within this method. It is just a callback - so better express the contract within the method name(see point 1)

=> `afterEntitySelected()` or `afterEntitySelectedCallback()`. What ever floats your boat.